### ISSUE TITLE
uninitialized constant RSpec::Rails::SystemExampleGroup

### ISSUE THREAD

**Issue by mockdeep** [rspec/rspec-rails#1889]
When upgrading from rspec-rails `3.6.1` to `3.7.1`, I'm seeing the error below when I try to run a test. It is triggered on the line in our code `require 'rspec/rails'`. Rails version `5.1.4`, Ruby version `2.4.2`.

```
/home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails/configuration.rb:57:in `add_test_type_configurations': uninitialized constant RSpec::Rails::SystemExampleGroup (NameError)
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails/configuration.rb:132:in `initialize_configuration'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails/configuration.rb:152:in `<module:Rails>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails/configuration.rb:2:in `<module:RSpec>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails/configuration.rb:1:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-rails-3.7.1/lib/rspec/rails.rb:15:in `<top (required)>'
	from /home/fletch/Dropbox/projects/chalk/synchroform/spec/rails_helper.rb:35:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `block in requires='
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `each'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `requires='
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:112:in `block in process_options_into'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:111:in `each'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:111:in `process_options_into'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:21:in `configure'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:99:in `setup'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:86:in `run'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:71:in `run'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:45:in `invoke'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/exe/rspec:4:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/rspec:23:in `load'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/rspec:23:in `<main>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by JonRowe** [comment]
Are you trying to use system tests without puma or capybara?

**Comment by mockdeep** [comment]
No, this actually happens with just arbitrary tests, not a system or feature test. Here's part of our `rails_helper.rb` up to where it throws the error:

```ruby
require 'rubygems'
require 'webmock/rspec'
def mutant?
  ENV['MUTANT'] == 'true'
end

unless mutant?
  require 'simplecov'

  if ENV['CIRCLE_ARTIFACTS']
    dir = File.join('..', '..', '..', ENV['CIRCLE_ARTIFACTS'], 'coverage')
    SimpleCov.coverage_dir(dir)
    SimpleCov.minimum_coverage 100
  else
    SimpleCov.coverage_dir('coverage/simplecov')
  end
  SimpleCov.start :rails do
    add_filter 'lib/mailer_previews'
    add_filter 'app/controllers/test_images_controller.rb'
    add_filter 'lib/nodeify'
  end
end

# This file is copied to spec/ when you run 'rails generate rspec:install'
ENV['RAILS_ENV'] ||= 'test'
require File.expand_path('../../config/environment', __FILE__)
def spec_root
  @spec_root ||= Rails.root.join('spec')
end

FactoryGirl.definition_file_paths = [spec_root.join('factories')]
FactoryGirl.find_definitions
Spring.after_fork { FactoryGirl.reload }

require 'rspec/rails'
```

**Comment by KeithP** [comment]
This also happens for Rails version 5.1.4, Ruby version 2.4.1.
Example output on running rspec:

  ```NameError:
  uninitialized constant RSpec::Rails::SystemExampleGroup
  # ./spec/rails_helper.rb:5:in `<top (required)>'
  # ./spec/routing/mymodels_routing_spec.rb:1:in `<top (required)>'

  An error occurred while loading ./spec/routing/mymodels_routing_spec.rb.
  Failure/Error: require 'rspec/rails'```

**Comment by JonRowe** [comment]
@mockdeep can you try out the `fix_system_tests_when_not_loaded` branch from #1893 ?

**Comment by mockdeep** [comment]
@JonRowe no luck. I had some trouble getting it to bundle as it was looking for `rspec-core ~> 3.8`. After hard coding it in the gemspec at `3.7` I was able to bundle, But getting a new error when I run the tests. One thing to note is that we're locked at an older version of Capybara `2.7.1`. Still haven't been able to get over that Selenium/Firefox bump with their marionette driver.

```
/home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require': cannot load such file -- capybara/minitest (LoadError)
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/actionpack-5.1.4/lib/action_dispatch/system_test_case.rb:2:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/scratch/rspec-rails/lib/rspec/rails/example/system_example_group.rb:28:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/scratch/rspec-rails/lib/rspec/rails/example.rb:11:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/scratch/rspec-rails/lib/rspec/rails.rb:13:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
	from /home/fletch/Dropbox/projects/chalk/synchroform/spec/rails_helper.rb:35:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `require'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `block in requires='
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `each'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1455:in `requires='
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:112:in `block in process_options_into'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:111:in `each'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:111:in `process_options_into'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/configuration_options.rb:21:in `configure'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:99:in `setup'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:86:in `run'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:71:in `run'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:45:in `invoke'
	from /home/fletch/.rvm/gems/ruby-2.4.2/gems/rspec-core-3.7.0/exe/rspec:4:in `<top (required)>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/rspec:23:in `load'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/rspec:23:in `<main>'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
	from /home/fletch/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by JonRowe** [comment]
It might be worth opening that new error as an issue with Rails, that looks like a mismatch in capybara versions

**Comment by mockdeep** [comment]
Okay, I opened [an issue on Rails](https://github.com/rails/rails/issues/30952), but based on [this SO response](https://stackoverflow.com/a/45202150/372479) from @twalpole it doesn't look too promising. Looks like we may be stuck on RSpec 3.6 for the time being.

**Comment by JonRowe** [comment]
@mockdeep if theres a minimum version of capybara required we can add this to our logic for testing the integration, for now I've changed #1893 to rescue load errors when loading the rails side which might fix your second error, can you give it a go?

**Comment by mockdeep** [comment]
@JonRowe looks like that did the trick. Thanks! I appreciate the flexibility.

**Comment by samphippen** [comment]
This should have been fixed by #1893
